coderd/database/migrations/000313_workspace_app_statuses.up.sql

```diff
@@ -0,0 +1,28 @@
+CREATE TYPE workspace_app_status_state AS ENUM ('working', 'complete', 'failure');
+
+-- Workspace app statuses allow agents to report statuses per-app in the UI.
+CREATE TABLE workspace_app_statuses (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    -- The agent that the status is for.
+    agent_id UUID NOT NULL REFERENCES workspace_agents(id),
+    -- The slug of the app that the status is for. This will be used
+    -- to reference the app in the UI - with an icon.
+    app_id UUID NOT NULL REFERENCES workspace_apps(id),
+	-- workspace_id is the workspace that the status is for.
+	workspace_id UUID NOT NULL REFERENCES workspaces(id),
+    -- The status determines how the status is displayed in the UI.
+    state workspace_app_status_state NOT NULL,
+    -- Whether the status needs user attention.
+    needs_user_attention BOOLEAN NOT NULL,
```

Comment:
Alternatively, we can go all-in on the `state` and have `waiting_on_input` as a separate state here. Right now the schema is neither here nor there.

Reply:
I think with the above comments this makes more sense now. It seems likely we'll kinda need both... I didn't want to have both, but seems like it may be necessary.